### ISSUE TITLE
link to static property when there is instance property with same name

### ISSUE THREAD

**Issue by myphysicslab** [jleyba/js-dossier#44]
Suggestion for syntax to reference a static property when there is also an instance property with the same name:  `{@link #ClassName.propertyName}`

The Dossier readMe states:
> Dossier will favor instance over static properties when de-referencing a hash link.

Dossier is already creating `id` tags of the form `ClassName.propertyName` for static properties.  Here is an example where I'm currently able to reference such a case by writing an HTML anchor tag:

```html
+ Set default values for DisplayPolygon class properties such as
  <a href="#DisplayPolygon.fillStyle"><code>DisplayPolygon.fillStyle</code></a>,
 *before creating* the DisplayPolygon object.
```

The suggestion is to generate the above HTML when Dossier sees a "static link" where the class name precedes the property name with a dot, so the above would look like this:

```html
+ Set default values for DisplayPolygon class properties such as
  {@link #DisplayPolygon.fillStyle},
 *before creating* the DisplayPolygon object.
```

IMO, this style would be more clear in general to distinguish static class properties, even when there is not an instance property with the same name.

**Comment by jleyba** [comment]
Is `{@link ClassName.staticPropertyName}` not generating a link?

**Comment by myphysicslab** [comment]
No, that results in `<code>ClassName.staticPropertyName</code>`.

Here is an example:
```
+ Set default values for DisplayPolygon class properties such as
  {@link DisplayPolygon.fillStyle}
 * *before creating* the DisplayPolygon object.
```

results in:

```html
<p>Set default values for DisplayPolygon class properties such as
<code>DisplayPolygon.fillStyle</code>
<em>before creating</em> the DisplayPolygon object.</p>
```

Here is the definition of the static property:

```javascript
/** Default color; used when creating a DisplayPolygon, see {@link #fillStyle}.
* @type {string} a CSS3 color value
* @static
*/
DisplayPolygon.fillStyle = 'black';
```

The instance property definition:

```javascript
/**
* @param {!myphysicslab.lab.engine2D.Polygon} polygon
* @constructor
* @struct
* @implements {myphysicslab.lab.view.DisplayObject}
*/
myphysicslab.lab.view.DisplayPolygon = function(polygon) {
  /**
  * @type {!myphysicslab.lab.engine2D.Polygon}
  * @private
  */
  this.polygon_ = polygon;
  /**
  * @type {boolean}
  * @private
  */
  this.dragable_ = isFinite(polygon.getMass()) && polygon.getDragPoints().length > 0;
  /** The color and transparency to use when drawing (filling) the polygon;
  * a CSS3 color value, or empty string to not fill the polygon.
  * @type {string}
  */
  this.fillStyle = DisplayPolygon.fillStyle;
```

**Comment by myphysicslab** [comment]
I also tried using the full name of the class like 
```
 {@link myphysicslab.lab.view.DisplayPolygon.fillStyle}
```
this results in a link to the instance property, not to the static property.

**Comment by jleyba** [comment]
I can confirm the bug that `{@link foo.Bar.property}` will resolve to an instance property instead of the static property.

Regarding `{@link DisplayPolygon.fillStyle}`, that won't work because the name resolution isn't that smart. I am planning on overhauling how `@link` references are resolved as part of #13, so I will consider this case for that.

**Comment by jleyba** [comment]
By the way, there are some architectural changes blocking progress on #13. While that's been open for a while, that actually is my current focus.

**Comment by myphysicslab** [comment]
That's interesting, I didn't know about issue #13.   I've been using fully qualified long names in comments instead of the `goog.scope` short names, so that hasn't been a problem for me.  Using `goog.scope` short names might be nice.  However, I'm not sure if it would be as clear as seeing the full name in the docs.  Perhaps it would be OK to use the short name for things that are in the same "package" (namespace). OTOH, if there is a link on a short name I guess the user can hover or click on it to figure out the long name.

**Comment by myphysicslab** [comment]
Confirmed fixed in 96d997e9b255874a9de276912460f9c6912b948b.  Using the full class name works as in `{@link myphysicslab.lab.view.DisplayPolygon.fillStyle}`
